test/bigtest/network/scenarios/uploading-files.js

```diff
@@ -0,0 +1,44 @@
+import { faker } from '@bigtest/mirage';
+
+export default server => {
+  let fileDefinitions;
+  let currentFileIndex = 0;
+
+  server.get('/data-import/uploadDefinitions');
+  server.post('/data-import/uploadDefinitions', (schema, request) => {
+    const { fileDefinitions: files } = JSON.parse(request.requestBody);
+
+    fileDefinitions = files.map(file => ({
+      id: faker.random.uuid(),
+      status: 'NEW',
+      ...file,
+    }));
+    const { attrs } = server.create('upload-definition', {
+      fileDefinitions,
+    });
+
+    return attrs;
+  }, 201);
+  server.post(
+    '/data-import/uploadDefinitions/:uploadDefinitionId/files/:fileId',
+    () => {
+      const file = fileDefinitions[currentFileIndex];
+
+      if (!file) {
+        return null;
+      }
+
+      file.uploadedDate = new Date().toString();
+      file.status = 'UPLOADED';
+      currentFileIndex++;
+
+      const { attrs } = server.create('upload-definition', {
```

Comment:
I think existing upload definition can be just updated instead of creating the new one?